application_sandbox/inline_uniform_block/main.cpp

```diff
@@ -0,0 +1,526 @@
+        nullptr,                                    // pNext
+        0,                                          // flags
+        *render_pass_,                              // renderPass
+        1,                                          // attachmentCount
+        &raw_view,                                  // attachments
+        app()->swapchain().width(),                 // width
+        app()->swapchain().height(),                // height
+        1                                           // layers
+    };
+
+    ::VkFramebuffer raw_framebuffer;
+    app()->device()->vkCreateFramebuffer(
+        app()->device(), &framebuffer_create_info, nullptr, &raw_framebuffer);
+    frame_data->framebuffer_ = containers::make_unique<vulkan::VkFramebuffer>(
+        data_->allocator(),
+        vulkan::VkFramebuffer(raw_framebuffer, nullptr, &app()->device()));
+  }
+
+  virtual void Update(float time_since_last_render) override {
+    model_data_->data().transform =
+        model_data_->data().transform *
+        Mat44::FromRotationMatrix(
+            Mat44::RotationX(3.14f * time_since_last_render) *
+            Mat44::RotationY(3.14f * time_since_last_render * 0.5f));
+  }
+
+  void PrepareCommandBuffer(size_t frame_index, CubeFrameData* frame_data) {
+    (*frame_data->command_buffer_)
+        ->vkBeginCommandBuffer((*frame_data->command_buffer_),
+                               &sample_application::kBeginCommandBuffer);
+    vulkan::VkCommandBuffer& cmdBuffer = (*frame_data->command_buffer_);
+
+    VkClearValue clear;
+    vulkan::MemoryClear(&clear);
+
+    VkRenderPassBeginInfo pass_begin = {
+        VK_STRUCTURE_TYPE_RENDER_PASS_BEGIN_INFO,  // sType
+        nullptr,                                   // pNext
+        *render_pass_,                             // renderPass
+        *frame_data->framebuffer_,                 // framebuffer
+        {{0, 0},
+         {app()->swapchain().width(),
+          app()->swapchain().height()}},  // renderArea
+        1,                                // clearValueCount
+        &clear                            // clears
+    };
+
+    cmdBuffer->vkCmdBeginRenderPass(cmdBuffer, &pass_begin,
+                                    VK_SUBPASS_CONTENTS_INLINE);
+
+    cmdBuffer->vkCmdBindPipeline(cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
+                                 *cube_pipeline_);
+    cmdBuffer->vkCmdBindDescriptorSets(
+        cmdBuffer, VK_PIPELINE_BIND_POINT_GRAPHICS,
+        ::VkPipelineLayout(*pipeline_layout_), 0, 1,
+        &frame_data->cube_descriptor_set_->raw_set(), 0, nullptr);
+    cube_.Draw(&cmdBuffer);
+    cmdBuffer->vkCmdEndRenderPass(cmdBuffer);
+
+    (*frame_data->command_buffer_)
+        ->vkEndCommandBuffer(*frame_data->command_buffer_);
+  }
+
+  void UpdateInlineUniformBlock(size_t frame_index, CubeFrameData* frame_data) {
+    frame_data->alpha_ =  static_cast<float>(rand()) / static_cast<float>(RAND_MAX);
```

Comment:
rand() is a little jarring.  Try this:

  int frame_count = 0;

  void UpdateInlineUniformBlock(size_t frame_index, CubeFrameData* frame_data) {
    frame_data->alpha_ =  static_cast<float>(std::abs(((frame_count + 30) % 60) - 30)) / 30.0f;
    frame_count++;

Reply:
thanks for this, this is indeed a better way.